chapters/ch22.md

```diff
@@ -1,80 +1,109 @@
 # 22. Comparing to the optimal error rate
 ->
-
+# 22. So sánh với tỉ lệ lỗi tối ưu 
 In our cat recognition example, the “ideal” error rate—that is, one achievable by an “optimal” classifier—is nearly 0%. A human looking at a picture would be able to recognize if it contains a cat almost all the time; thus, we can hope for a machine that would do just as well.
 ->
+Trong ví dụ nhận dạng mèo của chúng ta, tỉ lệ lỗi “lý tưởng” -- tỉ lệ có thể đạt được bởi một bộ phân loại “tối ưu” -- là gần với 0%. Gần như mọi lúc, một người nhìn vào bức ảnh có thể nhận ra có mèo trong đó hay không; do đó chúng ta có thể hy vọng máy móc cũng làm được điều tương tự. 
 
 Other problems are harder. For example, suppose that you are building a speech recognition system, and find that 14% of the audio clips have so much background noise or are so unintelligible that even a human cannot recognize what was said. In this case, even the most “optimal” speech recognition system might have error around 14%.
 ->
+Một số bài toán khác thì khó hơn. Ví dụ như chúng ta xây dựng một hệ thống nhận dạng giọng nói và nhận ra rằng 14% các clip âm thanh có quá nhiều nhiễu nền hoặc khó hiểu tới mức ngay cả con người cũng không thể nhận ra những gì được nói tới. Trong trường hợp này, ngay cả hệ thống “tối ưu” nhất cũng có thể có lỗi khoảng 14%. 
 
 Suppose that on this speech recognition problem, your algorithm achieves:
 ->
+Giả sử rằng với bài toán nhận dạng giọng nói này, thuật toán của bạn đạt được:
 
 * Training error = 15%
 ->
+* Lỗi huấn luyện = 15%
 
 * Dev error = 30%
 ->
+* Lỗi phát triển = 30%
 
 The training set performance is already close to the optimal error rate of 14%. Thus, there is not much room for improvement in terms of bias or in terms of training set performance. However, this algorithm is not generalizing well to the dev set; thus there is ample room for improvement in the errors due to variance.
 ->
+Hiệu quả trên tập huấn luyện đã gần với tỉ lệ lỗi tối ưu là 14%. Do đó, không có nhiều chỗ để cải thiện độ chệch  hoặc hiệu quả trên tập huấn luyện. Tuy nhiên, thuật toán này không tổng quát hoá tốt trên tập phát triển, do đó có rất nhiều chỗ để cải thiện lỗi do phương sai. 
```

Comment:
Cụm "room for improvement" dễ hiểu mà khá khó dịch nhỉ.

Reply:
công nhận là nhiều cái khi đặt tay xuống bàn phím dịch thì thấy khó vx.